cfme/common/provider.py

```diff
@@ -219,6 +219,17 @@ def _fill_provider_attributes(self, provider_attributes):
         if getattr(self, "project", None):
             provider_attributes["project"] = self.project
 
+        # Keystone API
```

Comment:
**REQUIRED:** We shouldn't have any provider specific information in here. If we need to do anything special for a provider, we should create a method on the provider wit ha default on the base so that we can do provider specific operations. We already do this for other things.

Reply:
PS. I know we have done this and allowed it in the past, but we need to make providers totally modular so there is no core contamination.